src/realm/object-store/object_store.cpp

```diff
@@ -826,51 +826,14 @@ static void apply_post_migration_changes(Group& group, std::vector<SchemaChange>
 
         void operator()(ChangeTableType op)
         {
-            if (handle_backlinks_automatically)
-                post_migration_embedded_objects_backlinks_handling(op.object);
-            table(op.object).set_table_type(static_cast<Table::Type>(*op.new_table_type));
+            table(op.object).set_table_type(static_cast<Table::Type>(*op.new_table_type),
+                                            handle_backlinks_automatically);
         }
         void operator()(RemoveTable) {}
         void operator()(ChangePropertyType) {}
         void operator()(MakePropertyNullable) {}
         void operator()(MakePropertyRequired) {}
         void operator()(AddProperty) {}
-
-        void post_migration_embedded_objects_backlinks_handling(const ObjectSchema* object_schema)
```

Comment:
This is now in `set_embedded()`. This eliminates some duplicated logic and is a more appropriate place for it, as the code here should just be calling the applicable schema change functions.